source/rust_verify/tests/examples.rs

```diff
@@ -56,12 +56,19 @@ fn run_example_for_file(file_path: &str) {
     };
 
     let mut mode = Mode::ExpectSuccess;
+    let mut deprecated_enhanced_typecheck = "";
+    if file_path.starts_with("example/state_machines") {
+        deprecated_enhanced_typecheck = "--deprecated-enhanced-typecheck";
+    }
 
     if let ["//", "rust_verify/tests/example.rs", command] = &first_line_elements[..] {
         match command.strip_suffix("\n").unwrap_or("unexpected") {
             "expect-success" => mode = Mode::ExpectSuccess,
             "expect-errors" => mode = Mode::ExpectErrors,
             "expect-failures" => mode = Mode::ExpectFailures,
+            "deprecated-enhanced-typecheck" => {
+                deprecated_enhanced_typecheck = "--deprecated-enhanced-typecheck";
+            }
```

Comment:
could you clean this up? the other 3 cases each set the `mode` variable, while this one leaves it as the default value, set several lines above.

Reply:
can you also please add a TODO/REVIEW to remove this option once we've removed `--deprecated-enhanced-typecheck`?